cpp/cmake/bob.cmake

```diff
@@ -17,13 +17,13 @@ endfunction(bob_always_full_rpath)
 
 function(bob_cmake_arg2 var type default)
   if(NOT ${var} STREQUAL "${default}")
-    if(${PROJECT_NAME}_CMAKE_ARGS)
+    if(${BBPCC_VAR_PREFIX}_CMAKE_ARGS)
```

Comment:
are you sure the variables in `bob.cmake` need to be renamed?

Reply:
You originally documented the prefix variable in the same section than the CMake options to control the compilation flags, so I assumed that you want to use this custom prefix for these variables as well: ${PREFIX}_CXX_OPTIMIZE, ${PREFIX}_CXX_SYMBOLS, ${PREFIX}_CXX_WARNINGS, ...

Reply:
Generally speaking, **all** CMake CACHE variables defined by hpc-coding-conventions should use ${BBPCC_VAR_PREFIX} otherwise, the users will face inconsistencies in CMakeCache.txt.